tiledb/ml/readers/_batch_utils.py

```diff
@@ -58,17 +70,48 @@ def __len__(self) -> int:
         Must be called after `set_batch_slice`.
         """
 
+    def _read_next_buffer_if_needed(
```

Comment:
@gsakkis Do we need extra unit tests for these?

Reply:
Strictly speaking we don't; this is a private method of a non-public class used by a non-public function, plus all new code is covered by higher level tests. Having said that, we could add specific tests for `BaseBatch` and its subclasses after the readers implementation has been stabilized.